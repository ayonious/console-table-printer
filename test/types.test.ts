import { COLOR, Table, ALIGNMENT } from '../index';

describe('Testing column alignment', () => {
  it('all kind of alignments are working', () => {
    // Create a table

    const red_color: COLOR = 'red';
    const green_color: COLOR = 'green';
    const left_alignment: ALIGNMENT = 'left';
    const center_alignment: ALIGNMENT = 'center';

    const p = new Table({
      columns: [
        {
          name: 'red_left_align_index',
          alignment: 'left',
          color: red_color,
        },
        { name: 'right_align_text', alignment: left_alignment },
        {
          name: 'green_value_center',
          alignment: center_alignment,
          color: green_color,
        },
      ],
    });

    // add rows with color
    p.addRow(
      {
        red_left_align_index: 2,
        right_align_text: 'This row is blue',
        green_value_center: 10.212,
      },
      { color: 'blue' }
    );
    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should allow chaining add methods', () => {
    const p = new Table()
      .addColumn('foo')
      .addColumns(['bar'])
      .addRow({ foo: '1' })
      .addRows([{ bar: '2' }]);

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
