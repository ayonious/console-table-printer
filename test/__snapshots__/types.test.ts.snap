// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing column alignment all kind of alignments are working 1`] = `
"┌──────────────────────┬──────────────────┬────────────────────┐
│[37m [0m[01mred_left_align_index[0m │[37m [0m[01mright_align_text[0m │[37m [0m[01mgreen_value_center[0m │
├──────────────────────┼──────────────────┼────────────────────┤
│[37m [0m[31m2                   [0m │[37m [0m[34mThis row is blue[0m │[37m [0m[32m      10.212      [0m │
└──────────────────────┴──────────────────┴────────────────────┘"
`;

exports[`Testing column alignment should allow chaining add methods 1`] = `
"┌─────┬─────┐
│[37m [0m[01mfoo[0m │[37m [0m[01mbar[0m │
├─────┼─────┤
│[37m [0m[37m  1[0m │[37m [0m[37m   [0m │
│[37m [0m[37m   [0m │[37m [0m[37m  2[0m │
└─────┴─────┘"
`;
